app/models/project.rb

```diff
@@ -33,10 +36,10 @@ def update_scores_and_images
   # Refresh raw data for all metrics
   def refresh_all_metrics!
     # Update (re-sample) all metrics for this project
-    ProjectMetric.metric_names.each do |metric|
+    ProjectMetrics.metric_names.each do |metric|
       # get credentials hash for this metric
-      credentials_hash = self.configs.where(:metric_name => metric).credentials
-      sample = ProjectMetric.class_for(metric).new(credentials_hash) # abstract factory
+      credentials_hash = self.configs.where(:metric_name => metric).first.options
+      sample = ProjectMetrics.class_for(metric).new(credentials_hash) # abstract factory
```

Comment:
@mtc2013 might be nice if `ProjectMetrics` had a class method that provided an iterator over metric names and coresponding class constants, eg:
```
ProjectMetrics.names_with_classes.each do |metric, klass|
  sample = klass.new(config_for(metric).options)
```